src/pykiso/lib/auxiliaries/zephyr.py

```diff
@@ -0,0 +1,223 @@
+
+        self.outdir = str(Path(test_directory).resolve() / "twister-out")
+        logging.info(f"Using Twister output directory: {self.outdir}")
+
+        self.process = subprocess.Popen(
+            [
+                self.twister_path,
+                "-vv",
+                "-T",
+                test_directory,
+                "-s",
+                testcase_name,
+                "--outdir",
+                self.outdir,
+                "--report-dir",
+                self.outdir,
+            ],
+            stderr=subprocess.PIPE,
+            shell=False,
+        )
+
+        if wait_for_start:
+            # Read twister output until test case was started
+            while True:
+                line = self._readline()
+                if line == None:
+                    return
+
+                if "OUTPUT: START - " in line:
+                    log.info(f"Zephyr test started.")
+                    return
+
+    def _readline(self) -> Optional[str]:
+        """Read a line from the twister process
+
+        :return: The line as string or None if the process has finished
+        """
+        while True:
+            read = self.process.stderr.readline()
+            if len(read) == 0:
+                return None
+            line = read.decode("utf-8").strip()
+            # Twister outputs many lines without a real message, remove those
+            output = line.find("- OUTPUT:")
+            if output != -1:
+                message = line[output + 9 :]
+                if len(message) == 0 or message.isspace():
+                    continue
+            log.debug(f"Twister: {line}")
+            return line
+
+    def _parse_xunit(self, file_path: str) -> TestResult:
+        """Parse xunit file
+
+        :return: The result of a single test case
+        """
+        tree = ET.parse(file_path)
+        testcase = tree.getroot().find("testsuite").find("testcase")
+        failure = testcase.find("failure")
+        skipped = testcase.find("skipped")
+        error = testcase.find("error")
+        if failure is not None:
+            log.info(f"Zephyr test failed: {failure.text}")
+            result = TestResult.FAILED
+        elif skipped is not None:
+            log.info(f"Zephyr test was skipped.")
+            result = TestResult.SKIPPED
+        elif error is not None:
+            log.error(f"Zephyr test failed with error: {error.text}")
+            result = TestResult.ERROR
+        else:
+            log.info(f"Zephyr test passed.")
+            result = TestResult.PASSED
+        return result
+
+    def wait_test(self) -> TestResult:
+        """Wait for the test to finish
+
+        :return: The result of the test, taken from the xunit output of twister
+        """
+        if self.process is None:
+            raise ZyphyrException("Twister was not started so can not wait.")
+
+        # Read the twister console for log output
+        while True:
```

Comment:
while block looks unnecessary 
ret = self.process.wait()  does actually what you want.

Reply:
As stderr is redirected and I don´t know how the output is buffered, the loop reads from stderr to prevent the twister process from blocking. Is there another way to guarantee that?